### PR TITLE
KotlinX.Coroutines 1.4.1

### PR DESCRIPTION
**Pull request by antonioseric**

### CODE REVIEW

**Comment by antonioser** [comment]
waiting #987 to be merged

**Comment by moljac** [comment]
Must check possible duplicate:

https://github.com/xamarin/XamarinComponents/pull/1053

**Review by moljac — CHANGES_REQUESTED** [review]
It would be good to prefix nugets/namespaces/assemblies with Xamarin if author is Microsoft.

**Comment by moljac** [comment]
@antonioseric @antonioser 

Antonio

I had to bind really fast 2 new bindings for 4-5 new AndroidX artifacts (`androidx.window` and `androidx.work` groups)

https://github.com/xamarin/XamarinComponents/pull/1053

Would it be possible to add remaining stuff to Kotlin folder? IMO this will be most likely updated all together (though not sure)

**Comment by antonioseric** [comment]
I prefer not to put in Kotlin folder KotlinX is released separately. I think it is better to be decoupled. I change rootnamespace and binding namespace.

Currently working on sample app (plan to finish it today)

**Comment by moljac** [comment]
Don't forget nuget packagenames prefix please

**Comment by moljac** [comment]
OK.  I will publish 1.3.4 from other PR and then delete it afterwards.

**Comment by 4brunu** [comment]
Any news on this? 😊

**Comment by antonioseric** [comment]
@moljac can we merge this and publish nugets?

**Comment by 4brunu** [comment]
Any news on this? 
@moljac @jpobst @mattleibow

**Review by moljac — APPROVED** [review]

File: Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Android/Transforms/Metadata.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿<metadata>
+
+    <attr path="/api/package[@name='kotlinx.coroutines.android']" name="managedName">Xamarin.KotlinX.Coroutines.CoroutinesAndroid</attr>
```
Comment (jpobst) [inline Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Android/Transforms/Metadata.xml]: I think we need to pick a standard capitalization for `Kotlinx`/`KotlinX` and be consistent with it.

File: Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Core.Jvm/Kotlinx.Coroutines.Core.Jvm.csproj [hunk 2]
```diff
@@ -0,0 +1,55 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (jpobst) [inline Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Core.Jvm/Kotlinx.Coroutines.Core.Jvm.csproj]: I don't think we want to bind these libraries that aren't intended for Android:
- Xamarin.Kotlinx.Coroutines.Core.Jvm
- Xamarin.Kotlinx.Coroutines.Core.Jdk8
- Xamarin.Kotlinx.Coroutines.Core.Reactive
- Xamarin.Kotlinx.Coroutines.Core.Rx2

Today it looks like we only publish 2 packages:
- Xamarin.KotlinX.Coroutines.Core
- Xamarin.KotlinX.Coroutines.CoroutinesAndroid

https://www.nuget.org/packages?q=xamarin+kotlin+coroutine

**Review by jpobst — COMMENTED** [review]
Namespace that is not renamed or hidden:
`### New Namespace Kotlin.Coroutines.Jvm.Internal`

File: Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Android/Transforms/Metadata.xml (hunk 1 above)
Comment (moljac) [reply Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Android/Transforms/Metadata.xml]: Changing all to KotlinX 

https://www.nuget.org/packages?q=Kotlinx

Comment (jpobst) [reply Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Android/Transforms/Metadata.xml]: Looks like our existing packages have already committed to `KotlinX`:
https://www.fuget.org/packages/Xamarin.KotlinX.Coroutines.Core/1.3.4

😉

File: Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Core.Jvm/Kotlinx.Coroutines.Core.Jvm.csproj (hunk 2 above)
Comment (moljac) [reply Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Core.Jvm/Kotlinx.Coroutines.Core.Jvm.csproj]: > Today it looks like we only publish 2 packages:

I think I bound those from Kotlin folder, just minimal dependencies I needed for GPS-FB-MLKit.

So, deleting or not publishing other packages?

Comment (moljac) [reply Android/KotlinxCoroutines/source/Kotlinx.Coroutines.Core.Jvm/Kotlinx.Coroutines.Core.Jvm.csproj]: Dependencies listed:

*   `kotlinx-coroutines-android`

    *   dependencies

        *   `kotlin-stdlib`

        *   `kotlinx-coroutines-core-jvm`        

*   `kotlinx-coroutines-core-jvm`

    *   dependencies

        *   `kotlin-stdlib`

        *   `kotlin-stdlib-common`

*   `kotlinx-coroutines-jdk8`

    *   dependencies

        *   `kotlin-stdlib`

        *   `kotlinx-coroutines-core-jvm`        

*   `kotlinx-coroutines-reactive`

    *   dependencies

        *   `kotlin-stdlib`

        *   `kotlinx-coroutines-core-jvm`

        *   `org.reactivestreams:reactive-streams`

*   `kotlinx-coroutines-rx2`

    *   dependencies

        *   `kotlin-stdlib`

        *   `kotlinx-coroutines-core-jvm`

        *   `io.reactivex.rxjava2:rxjava`        

        *   `kotlinx-coroutines-reactive`

Already bound:

*  `Kotlin.StdLib`
*  `Xamarin.Android.ReactiveStreams`
*  `RxJava`

So seem like those could be used on Android and it

**Comment by moljac** [comment]
because of conflicts manual merge had to be performed. 

branch: https://github.com/xamarin/XamarinComponents/tree/antonioseric-KotlinxCoroutines-1.4.1

PR: https://github.com/xamarin/XamarinComponents/pull/1155

**Comment by moljac** [comment]
closing this PR. Published from other PR/branch after manual merging

**Pull request closed without merging**